solver/simple.go

```diff
@@ -281,66 +297,66 @@ func (s *simpleSolver) preprocessInputs(ctx context.Context, st *state, vertex V
 				bklog.G(ctx).Warnf("failed to compute digest: %v", err)
 				return nil, err
 			} else {
-				scm.deps[i].computed = compDigest.String()
+				scm.deps[i].computed = compDigest
 			}
 		}
 
 		// Add input references to the struct as to link dependencies.
-		scm.inputs[i] = in.Vertex.Digest().String()
+		scm.inputs[i] = in.Vertex.Digest()
 
 		// Add the cached result to the input set. These inputs are used to
 		// reconstruct dependencies (mounts, etc.) for a new container run.
 		inputs = append(inputs, res)
 	}
 
-	s.cacheKeyManager.add(vertex.Digest().String(), &scm)
+	s.cacheKeyManager.add(vertex.Digest(), &scm)
 
 	return inputs, nil
 }
 
 type cacheKeyManager struct {
-	cacheMaps map[string]*simpleCacheMap
+	cacheMaps map[digest.Digest]*simpleCacheMap
 	mu        sync.Mutex
 }
 
 type cacheMapDep struct {
-	selector string
-	computed string
+	selector digest.Digest
```

Comment:
I switched to `digest.Digest` in a few places because using strings were leading to annoying conversions. They're just strings with some helper methods.